### DISCUSSION TITLE
OrbitControls tweaking

### DISCUSSION THREAD

**Discussion by GMezWheel** [zauberzeug/nicegui#1527] [Q&A] [answered]
I want to enable/disable some OrbitControls parameters from ui.scene:
- enableRotate Left
- enableRotate Up 
- enableZoom
- enablePan

How can I do it?

**Comment by falkoschindler** [comment, marked as answer]
Hi @GMezWheel!

You can access the OrbitControls via JavaScript like this:
```py
with ui.scene() as scene:
    scene.sphere()

ui.button('Enable zoom', on_click=lambda: ui.run_javascript(f'getElement({scene.id}).controls.enableZoom = true'))
ui.button('Disable zoom', on_click=lambda: ui.run_javascript(f'getElement({scene.id}).controls.enableZoom = false'))
```

**Reply by GMezWheel** [reply]
Hi @falkoschindler,

Thanks a lot. It works like a cham !

Is it possible to disable only 3D rotation in order to keep my 2D scene view? 
=> rotateUp : no
=> rotateLeft : yes

**Reply by falkoschindler** [reply]
For a 2D view it looks like you can set `maxPolarAngle = 0`.
I found it by looking into [OrbitControls.js](https://github.com/zauberzeug/nicegui/blob/main/nicegui/elements/lib/three/modules/OrbitControls.js).
